plugin/hotswap-agent-spring-plugin/src/main/java/org/hotswap/agent/plugin/spring/files/PropertyReload.java

```diff
@@ -0,0 +1,194 @@
+import org.springframework.beans.factory.annotation.AnnotatedBeanDefinition;
+import org.springframework.beans.factory.annotation.AnnotatedGenericBeanDefinition;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.beans.factory.config.PlaceholderConfigurerSupport;
+import org.springframework.beans.factory.config.PropertyPlaceholderConfigurer;
+import org.springframework.beans.factory.support.DefaultListableBeanFactory;
+import org.springframework.beans.factory.support.GenericBeanDefinition;
+import org.springframework.beans.factory.support.RootBeanDefinition;
+import org.springframework.context.support.PropertySourcesPlaceholderConfigurer;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.MutablePropertySources;
+import org.springframework.core.env.PropertySource;
+import org.springframework.util.ClassUtils;
+
+import java.io.IOException;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.*;
+import java.util.HashSet;
+import java.util.Set;
+
+public class PropertyReload {
+    private static AgentLogger LOGGER = AgentLogger.getLogger(PropertyReload.class);
+
+
+    public static void reloadPropertySource(DefaultListableBeanFactory beanFactory) {
+        ConfigurableEnvironment environment = beanFactory.getBean(ConfigurableEnvironment.class);
+        if (environment != null) {
+            for (PropertySource<?> propertySource : environment.getPropertySources()) {
+                if (propertySource instanceof IResourcePropertySource) {
+                    try {
+                        ((IResourcePropertySource) propertySource).reloadPropertySource();
+                    } catch (IOException e) {
+                        LOGGER.error("reload property source error", e, propertySource.getName());
+                    }
+                }
+            }
+        }
+
+        String[] beanFactoryBeanNamesForTypes = beanFactory.getBeanNamesForType(PlaceholderConfigurerSupport.class);
+        if (beanFactoryBeanNamesForTypes != null) {
+            for (String beanFactoryBeanName : beanFactoryBeanNamesForTypes) {
+                PlaceholderConfigurerSupport placeholderConfigurerSupport = beanFactory.getBean(beanFactoryBeanName, PlaceholderConfigurerSupport.class);
+                try {
+                    updatePlaceholderConfigurerSupport(beanFactory, placeholderConfigurerSupport);
+                } catch (Exception e) {
+                }
+            }
+        }
+    }
+
+
+    private static void updatePlaceholderConfigurerSupport(DefaultListableBeanFactory beanFactory, PlaceholderConfigurerSupport placeholderConfigurerSupport) {
+        if (placeholderConfigurerSupport instanceof PropertySourcesPlaceholderConfigurer) {
+            PropertySourcesPlaceholderConfigurer propertySourcesPlaceholderConfigurer = (PropertySourcesPlaceholderConfigurer) placeholderConfigurerSupport;
```

Comment:
the logic in if branch looks complex, should we extract it into one seperated method, and put more comments to explain why we do this? plus, should we stick to using org.hotswap.agent.util.ReflectionHelper#getNoException(java.lang.Object, java.lang.Class<?>, java.lang.String)?

Reply:
continue to extract 'if' branch to refreshPropertySourcesPlaceholderConfigurer() and 'else' to refreshPropertyPlaceholderConfigurer()?

Reply:
This method is simple, I add some comments and I think it is not necessary to seperate the method.